src/oatpp-openssl/Connection.cpp

```diff
@@ -71,10 +68,8 @@ void Connection::ConnectionContext::init() {
         case SSL_ERROR_WANT_WRITE:
           break;
         default:
-          //std::string errStr = std::string(ERR_error_string(ERR_get_error(), nullptr));
-          //OATPP_LOGE("[oatpp::openssl::Connection::ConnectionContext::init()]", "Error. Handshake failed. code=%d, msg='%s'", err, errStr.c_str());
-          //throw std::runtime_error("[oatpp::openssl::Connection::ConnectionContext::init()]: Error. Handshake failed. " + errStr);
-          return;
+          ErrorStack::logErrors("[oatpp::openssl::Connection::ConnectionContext::init()]");
+          throw std::runtime_error("[oatpp::openssl::Connection::ConnectionContext::init()]: Error. Handshake failed.");
```

Comment:
If the certificate isn't valid or the host name does not match, this is where the failure will occur.  Log the OpenSSL error stack (there will typically be more than one error message)